glib/tests/clone.rs

```diff
@@ -490,12 +491,15 @@ fn test_clone_macro_body() {
 
 #[test]
 fn test_clone_macro_async_kinds() {
-    let v = Rc::new(1);
+    let v = Rc::new(RefCell::new(1));
 
     // This one is still a rust unstable feature.
     // let _closure = clone!(@weak v => async move || 0);
-    let _closure = clone!(@weak v => move || async move { 0 });
-    let _closure = clone!(@weak v => async move { 0 });
+    let closure = clone!(@weak v => move || async move { *v.borrow_mut() += 1; });
+    block_on(closure());
+    assert_eq!(*v.borrow(), 2);
+    block_on(clone!(@weak v => async move { *v.borrow_mut() += 1; }));
+    assert_eq!(*v.borrow(), 3);
```

Comment:
For closures we also don't check if they're actually callable and do the right thing. Can you add that in a separate commit so we can be sure?

Reply:
What do you mean?

Reply:
We have e.g. `let _closure = clone!(@weak v => @default-return { let x = 12; x + 2 }, move || 19);` above. This doesn't test anything other than that the macro doesn't produce broken code. It could also produce an empty closure and we wouldn't know from the test.

So instead, call the closure and check that it actually contains the code it should :)

Reply:
I see, adding it.

Reply:
I found bugs so let's merge this and I'll open another PR.